lib/raxx.ex

```diff
@@ -266,6 +266,21 @@ defmodule Raxx do
     %{message | headers: headers ++ [{name, value}]}
   end
 
+  @doc """
+  Sets multiple headers in one go
+
+  ## Examples
+
+      iex> request(:GET, "/")
+      ...> |> set_headers([{"referer", "example.com"}, {"Authorization", "Bearer 1234567890"}])
+      ...> |> Map.get(:headers)
+      [{"referer", "example.com"}, {"accept", "text/html"}, {"Authorization", "Bearer 1234567890"}]
+  """
+  def set_headers(message = %{headers: _headers}, []),
+    do: message
+  def set_headers(message = %{headers: _headers}, [{name, value} | rest] = header_list) when is_list(header_list),
```

Comment:
don't need a guard clause. header_list must be a list because it is matched to `[head | tail]`